doc/packet_trimming/packet-trimming-design.md

```diff
@@ -0,0 +1,783 @@
+[Figure 5: PT switch show flow](#figure-5-switch-trimming-show-flow)  
+
+## List of tables
+
+[Table 1: Frontend event logging](#table-1-frontend-event-logging)  
+[Table 2: Backend event logging](#table-2-backend-event-logging)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+
+When the lossy queue exceeds a buffer threshold, it drops packets without any notification to the destination host.
+
+When a packet is lost, it can be recovered through fast retransmission or by using timeouts.  
+Retransmission triggered by timeouts typically incurs significant latency.
+
+To help the host recover data more quickly and accurately, packet trimming is introduced.  
+This feature upon a failed packet admission to a shared buffer, will trim a packet to a configured size,  
+and try sending it on a different queue to deliver a packet drop notification to an end host.
+
+## 1.2 Requirements
+
+### 1.2.1 Functionality
+
+**This feature will support the following functionality:**
+1. Global switch trimming configuration with per buffer profile control
+2. ACL disable trimming control policy
+3. Port/Queue trimming statistics
+
+### 1.2.2 Command interface
+
+**This feature will support the following commands:**
+1. config: set switch trimming global configuration
+2. show: display switch trimming global configuration
+
+### 1.2.3 Error handling
+
+#### 1.2.3.1 Frontend
+
+**This feature will provide error handling for the next situations:**
+1. Missing parameters
+2. Invalid parameter value
+
+#### 1.2.3.2 Backend
+
+**This feature will provide error handling for the next situations:**
+1. Missing parameters
+2. Invalid parameter value
+3. Parameter removal
+4. Configuration removal
+
+### 1.2.4 Event logging
+
+#### 1.2.4.1 Frontend
+
+**This feature will provide event logging for the next situations:**
+1. Switch trimming update
+
+###### Table 1: Frontend event logging
+
+| Event                           | Severity |
+|:--------------------------------|:---------|
+| Switch trimming update: success | NOTICE   |
+| Switch trimming update: error   | ERROR    |
+
+#### 1.2.4.2 Backend
+
+**This feature will provide event logging for the next situations:**
+1. Missing parameters
+2. Invalid parameter value
+3. Parameter removal
+4. Configuration removal
+5. Switch trimming update
+6. Buffer profile trimming update
+
+###### Table 2: Backend event logging
+
+| Event                                   | Severity |
+|:----------------------------------------|:---------|
+| Missing parameters                      | ERROR    |
+| Invalid parameter value                 | ERROR    |
+| Parameter removal                       | ERROR    |
+| Configuration removal                   | ERROR    |
+| Switch trimming update: success         | NOTICE   |
+| Switch trimming update: error           | ERROR    |
+| Buffer profile trimming update: success | NOTICE   |
+| Buffer profile trimming update: error   | ERROR    |
+
```

Comment:
Can you define the scope of the support and/or notable constraints? Eg, 

1) Trimming support is expected for both IPv4 and IPv6 frames? 

2) Does it support IP multicast frames? Would trimmed pkts be replicated and reach all receivers for retransmission request? 

3) What if the pkt to trim is destined for local tunnel decapsulation? Would it trim from the original frame or decapsulated frame?

Reply:
@adyeung the quick answer: all these things are vendor specific behavior. The current plan is not to extend the scope section. It has nothing to do with a trimming configuration. For the first question the answer is yes. Both IPv4/IPv6 supports trimming.